### ISSUE TITLE
Spectator slots block people from joining the game

### ISSUE THREAD

**Issue by AoAGeneral** [OpenRA/OpenRA#7810]
In CNC95 I have recently come across a strange glitch.

When people go into the spectator slot some players cannot join until the spectators are kicked. This has been tested with other players as well just to confirm this bug.
##

**Comment by penev92** [comment]
I'm really hoping you're not referring to the original game.
We may need some more info in order to reproduce this. Player names might help.

**Comment by AoAGeneral** [comment]
Not referring to the original no. Just the TD Mod of OpenRA.

Myself, Norman, Kutasani, bfg9000, yes no. The other two I can't remember the names.

**Comment by ChaoticMind** [comment]
This is a pretty "serious" bug that affects people in the lobby very frequently causing hosts to often automatically kick/disallow spectators just to avoid it.

I tried reproducing it but failed to do so in a consistent way. It has also happened when I hosted the game myself (instead of using the dedicated servers). It has only happened for me when there was at least one other non-spectator, but I can't be sure this is a prerequisite.

Sometimes, in a single game, if there's one spectator, a player manages to join, but another doesn't. I'm not sure based on what it works for some people but not others.

When a player attempts to join a game with a spectator (and hits the bug), it usually says "server is not responding" and you have the option to retry/cancel. 
From within the lobby, when a player attempts to join, the chat area shows that "player x has disconnected" without them ever joining, so people in the lobby are aware that someone is trying to join (and are hitting the bug). As a workaround, people usually go away from the spectator slot until everyone joins, then go back to the spectator slot (or the spectators get kicked).

Until someone finds a way to reproduce this consistently, I'm hoping this was useful. I think if a bunch of people/clients (3-4) try to experiment with it, it is very likely to be observed.

This is probably the bug that affects most people right now, so it would be great if it can be fixed before the next release.
